examples/khr-3hv/khr-3hv_continuous/controllers/supervisorsManager/khr3hvEnv.py

```diff
@@ -0,0 +1,206 @@
+        self.t = 5
+        # Number of observations
+        self.numObs = 10
+        # Total observations
+        self.obs = np.zeros(self.t * self.numObs)
+        # Lower and maximum values on observation space
+        lowObs = -np.inf * np.ones(self.t * self.numObs)
+        maxObs = np.inf * np.ones(self.t * self.numObs)
+        self.observation_space = Box(low=lowObs, high=maxObs, dtype=np.float64)
+        # Lower and maximum values on action space
+        lowAct = -2.356 * np.ones(6)
+        maxAct = 2.356 * np.ones(6)
+        self.action_space = Box(low=lowAct, high=maxAct, dtype=np.float64)
+
+        # Set up various robot components
+        self.motor_velocity = 2
+        self.robot = self.getSelf()
+        self.setup_agent()
+        self.motorPositionArr = np.zeros(6)
+        self.episodeScore = 0  # Score accumulated during an episode
+        self.prev_pos = self.robot.getPosition()[0]
+
+        # Logging parameters
+        self.distance_walk = 0
+        self.save_freq = wandbSaveFreq
+        self.counterLoging = 0
+        self.train = train
+        self.projectName = projectName
+        self.useRay = useRay
+
+    def get_observations(self):
+        """
+        Create the observation vector on each time step. 
+        The observations are the robot position (x,z,y), robot velocity, 
+        and the state of each motor.
+        :return: Observation vector
+        :rtype: numpy array
+        """
+        motorPos = self.robot.getPosition()
+        motorPos.append(self.robot.getVelocity()[0])
+
+        self.distance_walk = max(
+            self.distance_walk, self.robot.getPosition()[0])
+        self.counterLoging += 1
+        global GLOBAL_DISTANCE
+        GLOBAL_DISTANCE = max(GLOBAL_DISTANCE, self.robot.getPosition()[0])
+        # Logging on the wandb. When using Ray we don't use the custom logging because Ray has it own
+        # logging and produce compatiblity errors of two instaces of wandb.
+        if self.counterLoging % self.save_freq == 0 and self.train and not self.useRay:
+            #wandb.init(project=self.projectName, reinit=False)
+            wandb.log({"Episode distance walked": self.distance_walk,
+                       "Current x position": self.robot.getPosition()[0],
+                       "Globan distance walekd": GLOBAL_DISTANCE
+                       })
+
+        motorPos.extend([i for i in self.motorPositionArr])
+        # Shift the last 10 observations by one, on the time window of 5
+        self.obs = shift(self.obs, -self.numObs, cval=np.NaN)
+        self.obs[-self.numObs:] = motorPos
+
+        return np.array(self.obs)
+
+    def get_reward(self, action):
+        """
+        Calculate the reward of the agent. The reward award agent when moving forward on the x-axis.
+        Also, we are using a second term so the agent moves further away from his previous position.
+        :return: Reward value
+        :rtype: float
+
+        """
```

Comment:
This reward function seems a little bit weird to me. Maybe its the main reason for not converging well. Rewarding the agent based on the distance it moved from its previous position sounds good, but adding the x position straight onto the reward at each timesteps seems excessive. We can modify this reward function to reward the agent based on the difference of the robot's distance from the start position since the last step, if that makes sense. Basically keep the last distance `last_distance` it had from the center instead of its previous position and reward it based on `last_distance - current_distance`. Generally it seems similar to the pit escape problem where we need BB8 to move away from the center point.

```suggestion
        """
        Calculate the reward of the agent. Reward the agent when moving forward on the x-axis, but
        reward the agent based on the distance it moved from its previous position.
        :return: Reward value
        :rtype: float
        """
```